lib/crypto.js

```diff
@@ -276,13 +276,18 @@ Sign.prototype._write = function _write(chunk, encoding, callback) {
 
 Sign.prototype.update = Hash.prototype.update;
 
-Sign.prototype.sign = function sign(options, encoding) {
-  if (!options)
+Sign.prototype.sign = function sign(privateKey, encoding, options) {
+  if (!privateKey)
     throw new Error('No key provided to sign');
 
-  var key = options.key || options;
-  var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase);
+  if (typeof encoding == 'object' && !options) {
```

Comment:
if typeof encoding == object, && there _are_ also options, that should be an error